src/web/resources/report.js

```diff
@@ -35,6 +35,111 @@ function Element(tagname, props, children) {
+        const regressedTags = ["uni-start", "lt-target", "lt-start", 
+        "apx-start", "timeout", "crash", "error"]
+        const improvedTags = ["imp-start", "ex-start", "eq-start", "eq-target",
+        "gt-target"]
+        const advancedTags = []
+        advancedTags.push(regressedTags)
+        advancedTags.push(improvedTags)
+
+        // build labels
+        const regressedLabel = this.buildCheckboxLabel("regressed","Regressed",true)
+        const improvedLabel = this.buildCheckboxLabel("improved","Improved",true)
+
+        // add listeners
+        regressedLabel.addEventListener("click", this.attachLeaderToChildren("regressed",regressedTags))
+        improvedLabel.addEventListener("click", this.attachLeaderToChildren("improved",improvedTags))
+
+        const advancedDiv = [this.buildChildren(improvedTags),this.buildChildren(regressedTags)]
+
+        this.elt.appendChild(Element("div", [new Text("Filters"),]))
+        this.elt.appendChild(Element("div", [improvedLabel, regressedLabel]))
+        this.elt.appendChild(Element("details",[Element("summary","Advanced"),advancedDiv]))
+    },
+    buildCheckboxLabel: function(idTag, text, boolState) {
+        return Element("label", {id:idTag},
+            [Element("input",
+                { type: "checkbox", checked: boolState },
+                ""),
+            new Text(text)])
+    },
+    attachLeaderToChildren: function (leaderTag,listOfTags) {
+        return () => {
+            const improvedBox = document.querySelector(`#${leaderTag} input`)
+            listOfTags.forEach((str) => {
+                const currentCheckBox = document.querySelector(`#${str} input`)
+                currentCheckBox.checked = improvedBox.checked
+                this.updateDomNodesWithID(`${str}`, currentCheckBox.checked)
+            })
+        }
+    },
+    buildChildren: function (listOfTags) {
+        var childElements = []
+        // build child check boxes
+        listOfTags.forEach((child) => {
+            const count = document.querySelectorAll(`tr.${child}`)
+            const childBox = Element("label", {id: child}, 
+                [Element("input", { type: "checkbox", checked: true }, ""), new Text(`${this.rename(child)} (${count.length})`)])
+            // on click handler
+            childBox.addEventListener("click", () => {
+                const thisChild = document.querySelector(`#${child} input`)
+                this.updateDomNodesWithID(child, thisChild.checked)
+            })
+            childElements.push(childBox)
+        })
+        return childElements
+    },
+    // helper function to loop over the table and toggle state of nodes 
+    // with `#stringID`
+    updateDomNodesWithID: function (stringID, state) {
+        const siblingsList = document.querySelectorAll(`#results tbody tr`)
+        siblingsList.forEach((child, n, p) => {
+            if (child.classList.contains(stringID)) {
+                if (state) {
```

Comment:
You can do this but I think it'd be better to add or remove a class that then has the `display: none` feature in it. For one, that means you don't need to hardcode `table-row`.